projects/js-packages/components/components/contextual-upgrade-trigger/style.module.scss

```diff
@@ -1,24 +1,41 @@
 .cut {
-	border: 2px solid var(--jp-green-40);
-	border-radius: var(--jp-border-radius);
-	padding: calc( var(--spacing-base) * 2 ) calc( var(--spacing-base) * 3 );
+	border: 2px solid var( --jp-green-40 );
+	border-radius: var( --jp-border-radius );
+	padding: calc( var( --spacing-base ) * 2 ) calc( var( --spacing-base ) * 3 );
 	display: flex;
 	justify-content: space-between;
 	align-items: center;
-	gap: calc( var(--spacing-base) * 3 );
-	cursor: pointer;
+	gap: calc( var( --spacing-base ) * 3 );
 	text-align: left;
 	background: none;
 	margin: 0;
 	width: 100%;
+	position: relative;
 
-	&:focus, &:active {
-		outline-color: var(--jp-black);
+	button,
+	a {
+		all: unset;
+		color: var( --jp-black );
+		cursor: pointer;
+
+		&::after {
```

Comment:
What exactly we're doing here?

Reply:
With the previous setup, the entire component markup was wrapped in a `<button>` element, which is semantically incorrect because [buttons are only supposed to have phrasing content inside](https://www.w3.org/TR/2012/WD-html5-20120329/the-button-element.html#the-button-element).

By moving the `<button>` (or `<a>`) to the inside of the component it only contains the actual CTA part of the component. I've added a pseudo element that covers the entire component, so the whole thing is still clickable. It's the same principle as [Bootstrap's stretched link](https://getbootstrap.com/docs/5.1/helpers/stretched-link/).

As a bonus, it's also [a little bit more accessible](https://inclusive-components.design/cards/).

Reply:
> which is semantically incorrect because buttons are only supposed to have phrasing content inside

Nope, `div` has [no semantic](https://www.w3schools.com/html/html5_semantic_elements.asp) weight , and the `Text` / `Icon` returns a `p` and `svg` which are [phrasing content](https://www.w3.org/TR/2012/WD-html5-20120329/content-models.html#phrasing-content). So it follows all the semantics.

But got your point, it works too :)

Reply:
I think it's quite nice to be able to have an href rather than only use an onClick (although I realise we could still do that with the whole element being the button/anchor), and have the element only wrap the CTA. As it works shall we leave it like this for now?